<!-- 
Files tuned for ethzasl_icp_mapping commit 52d78c96c2cf2f58a400c649c561dbfec95e41a4
and for libpointmatcher commit e048c4d1180b1de005af072bf137ea4d8cf2b826

"Long-term 3D map maintenance in dynamic environments"
-->

<launch>
	<!--
	<arg name="file_path" default="/home/fabio/Documents/ASL-Lidar/data/velodyne/2016-03-11/2016-03-11-12-11-53.bag"/>
	-->
	

	
	<!-- <arg name="rviz_config" /> -->
	
	<!--
	<node pkg="rviz" type="rviz" name="visualization" output="screen" respawn="true" args="-d /home/fabio/Documents/ASL-Lidar/data/velodyne/rvis/velodyne16.rviz" />
	-->
	<node pkg="rviz" type="rviz" name="visualization" output="screen" respawn="true" args="-d /home/xiaqing/.rviz/dynamic_slam.rviz" />
<!--
    <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="0 0 0 0 0 0 1 odom velodyne 100" />
-->

	<param name="/use_sim_time" value="true"/>
	<node name="localization" type="localization" pkg="dynamic_SLAM" output="screen" >
		<param name="TASK" value="without_map"/>
		<param name="BeginID" value="0"/>
		<param name="firstPose" value="$(find dynamic_SLAM)/cfg/firstPose.txt" />
		<param name="Debug_ICP" value="False" />
		<param name="mapElevation" value="False" />
	
		<param name="subscribe_scan" value="false"/>
		<param name="subscribe_cloud" value="true"/>
		<remap from="scan" to="/scan"/>
		<param name="cloud_in" value="/velodyne_points"/>
		<param name="odom_msg_name" value="/thomas_robot/odom" />
		<param name="imu_msg_name" value="/mti/sensor/imu" />

		<param name="use_imu" value="true" />
		<param name="use_odom" value="false" />
		
		
		<param name="icpConfig" value="$(find dynamic_SLAM)/cfg/icp_dynamic.yaml" />
		<param name="inputFiltersConfig" value="$(find dynamic_SLAM)/cfg/input_filters.yaml" />
		<param name="mapPostFiltersConfig" value="$(find dynamic_SLAM)/cfg/map_post_filters.yaml" />
		<param name="maximumDistFiltersConfig" value="$(find dynamic_SLAM)/cfg/maximumDistFiltering.yaml" />
		

		<param name="odom_frame" value="/velodyne"/>
		<param name="map_frame" value="/map" />
		<param name="imu_frame" value="/imu" />

		<param name="useROSLogger" value="false" />
		<param name="inputQueueSize" value="1" />
		<param name="tfRefreshPeriod" value="0.005" />
		<param name="minOverlap" value="0.2" /> 
		<param name="maxOverlapToMerge" value="0.99" /> 
		<param name="minReadingPointCount" value="700" /> 
		<param name="minMapPointCount" value="10000" />
		<param name="localizing" value="true" /> 
		<param name="mapping" value="true" />

		<param name="priorStatic" value="0.45"/>
		<param name="priorDyn" value="0.55"/>
		<param name="maxAngle" value="0.01"/>
		<param name="eps_a" value="0.1"/> 
		<param name="eps_d" value="0.1"/>
		<param name="alpha" value="0.99"/>
		<param name="beta" value="0.90"/>
		<param name="maxDyn" value="0.90"/>
		<param name="maxDistNewPoint" value="0.7"/>
		<param name="sensorMaxRange" value="100.0"/>
		<param name="halfRoadWidth" value="3"/>
		<param name="MaxLocalMappoints" value="17000"/>
		<param name="KFthreOverlap" value="0.88"/>
		<param name="maxLocalMapLength" value="20"/>
		<param name="windows_width" value="7"/>
		<param name="delta_width" value="1"/>
		<param name="initial_time" value="5"/>
	</node>

</launch>
